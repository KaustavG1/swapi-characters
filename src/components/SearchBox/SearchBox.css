.search-container {
  outline: 1px solid red;
}

.search-bar {
  width: 90%;
}
