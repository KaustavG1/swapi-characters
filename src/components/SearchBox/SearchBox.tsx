import React from "react";
import "./SearchBox.css";

export interface SearchBoxProps {
  searchTerm: string;
  setSearchTerm: React.Dispatch<React.SetStateAction<string>>;
}

function SearchBox({searchTerm, setSearchTerm}: SearchBoxProps) {
  const handleInputChange = (event: React.FormEvent<HTMLInputElement>) => {
    setSearchTerm((event.target as HTMLInputElement).value);
  }

  return (
    <div className="search-container">
      <input
        className="search-bar"
        type="text"
        placeholder="search characters"
        value={searchTerm}
        onChange={handleInputChange}
      />
    </div>
  );
}

export default SearchBox;

