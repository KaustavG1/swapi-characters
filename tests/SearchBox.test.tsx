import { it, expect, vi } from "vitest";
import { fireEvent, render, screen } from "@testing-library/react";
import SearchBox from "../src/components/SearchBox/SearchBox";

describe("SearchBox component", () => {
  it("renders 'SearchBox' component", () => {
    const setSearchTerm = vi.fn();
    const { container } = render(
      <SearchBox searchTerm="test search term" setSearchTerm={setSearchTerm} />
    );
    const searchBar = screen.getByPlaceholderText(/search characters/i);
    expect(searchBar).toBeInTheDocument();
    fireEvent.change(searchBar, { target: { value: "new value" } });
    expect(setSearchTerm).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });
});
