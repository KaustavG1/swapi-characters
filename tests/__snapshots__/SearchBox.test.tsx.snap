// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBox component > renders 'SearchBox' component 1`] = `
<div>
  <input
    class="search-bar"
    placeholder="search characters"
    type="text"
    value="test search term"
  />
</div>
`;
